### PR TITLE
Feat/biovar map

### PR DESCRIPTION
**Pull request by kfilip94**
[PREVIEW](https://copernicus-forest-git-feat-biovar-map.vizzuality1.now.sh/SWE/bioclimatic/)
This PR introduces a few features to the `Bioclimatic variables` page:
- we can filter biovars layer, depending on the start/end year, scenario or country,
- timeline works,
- removes the "blink" effect when changing filters (now it refreshes the data props instead of reloading whole bioclimatic page component),
- add the legend to the map with a biovar name,
- moves zoom bar to the separate component, now we have the same version of this component in `Species distribution` and `Bioclimatic variables`
zub0n-wt23e

Small tweaks in `Map distribution` page:
- add download button in the right corner,
- center the ramp in the legend,
- change zoom in / zoom out icons
image

 **ISSUES:**
- I'm still not sure how I should calculate the legend buckets. At this moment I'm taking the minimum, medium and maximum temperature from **the selected biovar's chart data** and I'm passing this data to the layer config, to create temperature color buckets. Is it a correct approach? What is the difference between the data on the chart and on the map?
- It seems that we don't have the layer data for 1995 in rcp45 and rcp85 because as I understand 1995 is the `current` scenario, so the timeline shows 2020 as the first year when 1995 is selected on filters, is that fine?
image

### CODE REVIEW

**Review by edwardpmorris — CHANGES_REQUESTED** [review]
Nice. Good job!

> I'm still not sure how I should calculate the legend buckets. At this moment I'm taking the minimum, medium and maximum temperature from the selected biovar's chart data and I'm passing this data to the layer config, to create temperature color buckets. Is it a correct approach? What is the difference between the data on the chart and on the map?

Ideally we would like to have the min, median and maximum `wieghtedmean` from the map data layer for EACH biovar of EACH scenario for ALL time-intervals, i.e., rcp45-biovar01-1995--2090. Still, if it is not feasible or causes performance issues to calculate this from the map layer, the approach implemented seems good to me. An alternative could be that we calculate these in the data pipeline and add to the graphCMS data. No action required; advise on feasibility  

> It seems that we don't have the layer data for 1995 in rcp45 and rcp85 because as I understand 1995 is the current scenario, so the timeline shows 2020 as the first year when 1995 is selected on filters, is that fine?

- [ ]  Ideally, I think we should include 1995 time-interval on the map; in this case 1995 belongs to both scenarios. Please adjust or advise on if changes to the `scenario` schema are required.

- [x] Please change biovar 12 to 19 color ramp to precipitation 3 color ramp https://projects.invisionapp.com/share/T4U1GVW3DNE#/screens/386143376_UI_KIT

- [ ] Please check transparency of biovar 01 to 11 color ramp; on the map the visual effect seems to suggest transparency within the color ramp, which seems different to biovar19, for example.

Also at present a user is unable to know the values on the map, the scale bar is low to high. Please advise as to feasibility of including either values on the scale bar or allowing hover or pop-up interaction on the map to reveal values.

File: src/layers.js [hunk 1]
```diff
@@ -1,3 +1,40 @@
+export const COLOR_RAMP = [
```
Comment (dfrico) [inline src/layers.js]: which color ramp is this? Should we rename?

File: src/pages/bioclimatic/component.jsx [hunk 2]
```diff
@@ -1,37 +1,80 @@
-import React, { useState } from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
-import { LayerManager, Layer } from 'layer-manager/dist/components';
-import { PluginMapboxGl } from 'layer-manager';
+// import { LayerManager, Layer } from 'layer-manager/dist/components';
```
Comment (dfrico) [inline src/pages/bioclimatic/component.jsx]: are these comments for later use?

File: src/pages/bioclimatic/component.jsx [hunk 3]
```diff
@@ -53,23 +96,41 @@ function BioClimaticPage(props) {
...
+          </div>
           <Modal
             title="Bioclimatic variables data"
             text={`Bioclimatic variables derived from Copernicus describing temperature and precipitation annual tendencies, seasonality
                   and extreme climatic conditions, including a combination of both environmental factors for current and future scenarios.`}
             storageKey="bioclimatic"
           />
+          <RampLegend
+            title={biovarName}
+            colorRamp={['#FEF6B5', '#FFA679', '#E15383']} // purple
```
Comment (dfrico) [inline src/pages/bioclimatic/component.jsx]: maybe we should use vars for things like these 🤔

File: src/pages/bioclimatic/index.js [hunk 4]
```diff
@@ -53,12 +54,15 @@ const Container = () => {
     scenarios &&
     scenarios.reduce((acc, sc) => {
       const years = getYears(sc);
+      const startIndex = years.indexOf(startYear * 1);
+      const endIndex = years.indexOf(endYear * 1);
```
Comment (dfrico) [inline src/pages/bioclimatic/index.js]: `* 1` ?

**Review by dfrico — COMMENTED** [review]
Apart from @edwardpmorris 's comments and a couple of questions I have, the code looks good to me

File: src/layers.js (hunk 1 above)
Comment (kfilip94) [reply src/layers.js]: Ramps `COLOR_RAMP` and `BIOCLIMATIC_COLOR_RAMP` are not needed there, I messed this up while rebasing. I going to remove them, thanks :+1:

File: src/pages/bioclimatic/component.jsx (hunk 2 above)
Comment (kfilip94) [reply src/pages/bioclimatic/component.jsx]: No, we don't need them anymore, removed. Thanks :+1:

File: src/pages/bioclimatic/component.jsx (hunk 3 above)
Comment (kfilip94) [reply src/pages/bioclimatic/component.jsx]: Great idea! I moved them to `constants.js` file, together with Species distribution ramp :+1:

File: src/pages/bioclimatic/index.js (hunk 4 above)
Comment (kfilip94) [reply src/pages/bioclimatic/index.js]: It was my "creative" way of converting params in URL from string to number - already changed to `years.indexOf(Number(startYear))`  thanks :+1:

**Comment by kfilip94** [comment]
@edwardpmorris Thank you for your review, it's super helpful :+1: 
Referring to your comment:

-  "Ideally, I think we should include 1995 time-interval on the map; in this case 1995 belongs to both scenarios. Please adjust or advise on if changes to the scenario schema are required." 

I can include 1995 on the map, but the problem is that there is no data in carto db for rcp45 and rcp85 scenarios; check this sql query on [the example in carto](https://vizzuality-enterprise.carto.com/u/copernicus-forests/builder/711187bb-580c-4dd9-86eb-bf6b5a8f0ed0/layer/1b15cdb6-3e33-4c5b-8912-a50d8323de47/data)
`WITH a AS (SELECT cartodb_id, the_geom_webmercator, uuid, iso3 FROM all_geometry WHERE iso3 = 'SWE') SELECT a.the_geom_webmercator, a.cartodb_id, b.uuid, b.timeinterval as year, b.biovar, b.scenario, b.wieghtedmean FROM zonal_bv_uuid as b INNER JOIN a ON b.uuid = a.uuid WHERE timeinterval = 1995 AND biovar = 'biovar01' AND scenario = 'rcp85'`

Is it possible to add the data to these scenarios in carto db?

- "Please change biovar 12 to 19 color ramp to precipitation 3 color ramp" 
Done, thank you for spotting this :+1:  I didn't know that there should be a different ramp

- "Please check the transparency of biovar 01 to 11 color ramp; on the map, the visual effect seems to suggest transparency within the color ramp, which seems different to biovar19, for example."
I'm not sure if I understand this issue:
Untitled

The difference between smooth, orange transition in bio1 and solid transition in bio19 is caused by a relatively big range between min and max values and relatively small range in bio19. In this sense, it's not a bug, but it's directly tied with the buckets issue that we were talking about before. 
I'll try to improve bucketing as you suggested by playing with `wieghtedmean` values, but  I'm not sure if I can get access to these data from the layer manager level, I'll try to play with this and I'll let you know ;)

**Comment by kfilip94** [comment]
Ok, in terms of buckets for bioclimatic variables, I'd leave it as it is for now. I don't have access to the data of the layer from the Layer Manger level. Having this in GraphCMS could be a good solution but for now, I'd vote for merging this PR without changing buckets.

**Comment by sorodrigo** [comment]
I agree lets merge and maybe exploring getting the buckets from carto using one of these functions: 
https://github.com/CartoDB/cartodb-postgresql/wiki/CDB_QuantileBins
https://github.com/CartoDB/cartodb-postgresql/wiki/CDB_JenksBins

**Comment by kfilip94** [comment]
Hey @sorodrigo :wave:  I was talking about this with Ed one day and it seems that these functions are not available in the carto account that we have :(

**Comment by kfilip94** [comment]
@edwardpmorris @blayhem If you are ok with my comments/changes above, please approve this PR so I can merge it :wink:

**Review by dfrico — APPROVED** [review]

**Comment by dfrico** [comment]
Good job, let’s merge then!

**Pull request merged**